File: trackr-app/app/src/main/java/com/io/trackr/trackr_app/usecases/EventOutputData.java
package usecases;

import entities.Anniversary;
import entities.Event;

import java.time.LocalDate;
import java.util.Set;

/**
 * A dataclass representing the Data that Presenter will retrieve from EventManager
 */
public class EventOutputData {
    private EventTypes eventType;
    private String firstName;
    private String lastName;
    private String personDescription;
    private Set<String> tags;
    private LocalDate date;
    private LocalDate remindDeadline;

    public enum EventTypes {
        ANNIVERSARY,
        BIRTHDAY,
    }

    public EventOutputData(Event event) {
        if (event != null) {
            this.eventType = event instanceof Anniversary ? EventTypes.ANNIVERSARY : EventTypes.BIRTHDAY;
            this.firstName = event.getPerson().getFirstName();
            this.lastName = event.getPerson().getLastName();
            this.personDescription = event.getPerson().getDescription();
            this.tags = event.getPerson().getTags();
            this.date = event.getDate();
            this.remindDeadline = event.getReminderDeadline();
        }
    }

    public String getEventType() {
        return eventType.toString();
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getPersonDescription() {
        return personDescription;
    }

    public Set<String> getPersonTags() {
        return tags;
    }

    public LocalDate getDate() {
        return date;
    }

    public LocalDate getRemindDeadline() {
        return remindDeadline;
    }
}

File: trackrapp/app/src/main/java/com/trackr/trackr_app/usecases/EventOutputData.kt
package usecases

import entities.Anniversary

/**
 * A dataclass representing the Data that Presenter will retrieve from EventManager
 */
class EventOutputData(event: Event?) {
    private var eventType: EventTypes? = null
    var firstName: String? = null
    var lastName: String? = null
    var personDescription: String? = null
    var personTags: Set<String>? = null
    private var date: LocalDate? = null
    private var remindDeadline: LocalDate? = null

    enum class EventTypes {
        ANNIVERSARY, BIRTHDAY
    }

    fun getEventType(): String {
        return eventType.toString()
    }

    fun getDate(): LocalDate? {
        return date
    }

    fun getRemindDeadline(): LocalDate? {
        return remindDeadline
    }

    init {
        if (event != null) {
            eventType = if (event is Anniversary) EventTypes.ANNIVERSARY else EventTypes.BIRTHDAY
            firstName = event.getPerson().getFirstName()
            lastName = event.getPerson().getLastName()
            personDescription = event.getPerson().getDescription()
            personTags = event.getPerson().getTags()
            date = event.getDate()
            remindDeadline = event.getReminderDeadline()
        }
    }
}